scipy/optimize/tests/test_optimize.py

```diff
@@ -507,6 +507,23 @@ def test_ncg_negative_maxiter(self):
                                    args=(), options=opts)
         assert result.status == 1
 
+    def test_ncg_zero_xtol(self):
+        # Regression test for gh-20214
+        def cosine(x):
+            return np.cos(x[0])
+
+        def jac(x):
+            return -np.sin(x)
+
+        x0 = [0.1]
+        xtol = 0
+        result = optimize.minimize(cosine,
+                                   x0=x0,
+                                   jac=jac,
+                                   method="newton-cg",
+                                   options=dict(xtol=xtol))
+        assert result.status == 0
```

Comment:
Could we also test the value of the obtained solution?

Reply:
Good point, I've added that to the test.